recipes/freeimage/all/conanfile.py

```diff
@@ -0,0 +1,99 @@
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        tools.check_min_cppstd(self, "11")
+        if self.options.shared:
+            del self.options.fPIC
+        self.output.warn("TIFF and G3 plugins are disabled.")
+
+    def requirements(self):
+        self.requires("zlib/1.2.11")
+        self.requires("libjpeg/9d")
+        self.requires("openjpeg/2.3.1")
+        self.requires("libpng/1.6.37")
+        self.requires("libwebp/1.1.0")
+        self.requires("openexr/2.5.2 ")
+        self.requires("libraw/0.19.5")
+        self.requires("jxrlib/cci.20170615")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "FreeImage"
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.configure(build_dir=self._build_subfolder)
+        return self._cmake
+
+    def build(self):
+        # tools.rmdir(os.path.join(self._source_subfolder, "Source", "LibJPEG"))
+        tools.rmdir(os.path.join(self._source_subfolder, "Source", "LibPNG"))
+        tools.rmdir(os.path.join(self._source_subfolder, "Source", "LibTIFF4"))
+        # tools.rmdir(os.path.join(self._source_subfolder, "Source", "ZLib"))
+        tools.rmdir(os.path.join(self._source_subfolder, "Source", "LibOpenJPEG"))
+        tools.rmdir(os.path.join(self._source_subfolder, "Source", "LibJXR"))
+        tools.rmdir(os.path.join(self._source_subfolder, "Source", "LibWebP"))
+        tools.rmdir(os.path.join(self._source_subfolder, "Source", "LibRawLite"))
+        tools.rmdir(os.path.join(self._source_subfolder, "Source", "OpenEXR"))
+
+        for patch in self.conan_data.get("patches", {}).get(self.version, {}):
+            tools.patch(**patch)
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        cmake = self._configure_cmake()
+        cmake.install()
+        self.copy("license-fi.txt", dst="licenses", src=self._source_subfolder)
+        self.copy("license-gplv3.txt", dst="licenses", src=self._source_subfolder)
+        self.copy("license-gplv2.txt", dst="licenses", src=self._source_subfolder)
+
+    def package_info(self):
+        self.cpp_info.libs = tools.collect_libs(self)
+
+        self.cpp_info.names["pkg_config"] = "freeimage"
+        self.cpp_info.names["cmake_find_package"] = "FreeImage"
+        self.cpp_info.names["cmake_find_package_multi"] = "FreeImage"
```

Comment:
These names are not *official*, aren't they?
```suggestion
```

Reply:
That's right. I just use the most popular pattern. (Ex: https://github.com/OGRECave/ogre/blob/master/CMake/Packages/FindFreeImage.cmake)